scripts/in_container/run_provider_yaml_files_check.py

```diff
@@ -0,0 +1,482 @@
+
+
+def check_providers_are_mentioned_in_issue_template(yaml_files: dict[str, dict]):
+    prefix_len = len("apache-airflow-providers-")
+    short_provider_names = [d["package-name"][prefix_len:] for d in yaml_files.values()]
+    # exclude deprecated provider that shouldn't be in issue template
+    deprecated_providers: list[str] = []
+    for item in deprecated_providers:
+        short_provider_names.remove(item)
+    jsonpath_expr = parse('$.body[?(@.attributes.label == "Apache Airflow Provider(s)")]..options[*]')
+    with PROVIDER_ISSUE_TEMPLATE_PATH.open() as issue_file:
+        issue_template = yaml.safe_load(issue_file)
+    all_mentioned_providers = [match.value for match in jsonpath_expr.find(issue_template)]
+    try:
+        print(
+            f" -- Checking providers: present in code (left), "
+            f"mentioned in {PROVIDER_ISSUE_TEMPLATE_PATH} (right)"
+        )
+        assert_sets_equal(set(short_provider_names), set(all_mentioned_providers))
+    except AssertionError as ex:
+        print(ex)
+        sys.exit(1)
+
+
+def check_providers_have_all_documentation_files(yaml_files: dict[str, dict]):
+    expected_files = ["commits.rst", "index.rst", "installing-providers-from-sources.rst"]
+    for package_info in yaml_files.values():
+        package_name = package_info["package-name"]
+        provider_dir = DOCS_DIR.joinpath(package_name)
+        for file in expected_files:
+            if not provider_dir.joinpath(file).is_file():
+                errors.append(
+                    f"The provider {package_name} misses `{file}` in documentation. "
+                    f"Please add the file to {provider_dir}"
+                )
+
+
+if __name__ == "__main__":
+    architecture = Architecture().get_current()
+    console.print(f"Verifying packages on {architecture} architecture. Platform: {platform.machine()}.")
+    provider_files_pattern = pathlib.Path(ROOT_DIR).glob("airflow/providers/**/provider.yaml")
+    all_provider_files = sorted(str(path) for path in provider_files_pattern)
+
+    if len(sys.argv) > 1:
+        paths = sorted(sys.argv[1:])
+    else:
+        paths = all_provider_files
+
+    all_parsed_yaml_files: dict[str, dict] = _load_package_data(paths)
+
+    all_files_loaded = len(all_provider_files) == len(paths)
+    check_integration_duplicates(all_parsed_yaml_files)
+
+    check_duplicates_in_integrations_names_of_hooks_sensors_operators(all_parsed_yaml_files)
+
+    check_completeness_of_list_of_transfers(all_parsed_yaml_files)
+    check_duplicates_in_list_of_transfers(all_parsed_yaml_files)
+    check_hook_classes(all_parsed_yaml_files)
+    check_extra_link_classes(all_parsed_yaml_files)
+    check_correctness_of_list_of_sensors_operators_hook_modules(all_parsed_yaml_files)
+    check_unique_provider_name(all_parsed_yaml_files)
+    check_providers_are_mentioned_in_issue_template(all_parsed_yaml_files)
+    check_providers_have_all_documentation_files(all_parsed_yaml_files)
```

Comment:
Cross PR question, are this checks could potentially replace this test class
https://github.com/apache/airflow/pull/28459#discussion_r1052116651 ?

Maybe not now. Just wondering.

Reply:
I think those tests are a bit different - they are more for correctness of "provider.yam" - i.e. what is exposed to outside world, where the tests in #28469 are more about naming and structure of the provider packages (though I agree they should be trimmed a bit).